tasks/elasticsearch-RedHat-version-lock.yml

```diff
@@ -2,6 +2,26 @@
 - name: RedHat - install yum-version-lock
   become: yes
   yum: name=yum-plugin-versionlock state=present update_cache=yes
+
+- name: RedHat - check elasticsearch version lock
+  become: yes
+  shell: yum versionlock list | grep -c elasticsearch
+  register: es_version_locked
+  args:
+    warn: false
+  failed_when: False
+  changed_when: False
+
 - name: RedHat - lock elasticsearch version
   become: yes
   shell: yum versionlock delete 0:elasticsearch* ; yum versionlock add {{ es_package_name }}{% if es_version is defined and es_version != "" %}-{{ es_version }}{% endif %}
```

Comment:
Yes, it's necessary to delete old version locks, because they're version specific (obviously :smiley: ) and any existing lock would prevent us from upgrading ES package or setting new version locks during upgrade.

But thanks to pointing that out I've found different problem that would prevent users from performing upgrades via Ansible with version lock engaged. Fix is on the way.

Reply:
My question was more related to `es_version_locked.stdout|int == 0` part of the conditional.

`es_version_locked.stdout` is the result of `yum versionlock list | grep -c {{es_package_name}}-{{es_version}}` so it should be set to the number of elasticsearch packages currently present in versionlock.

As I understand this, If you run `yum versionlock delete 0:elasticsearch* ; yum versionlock add {{ es_package_name }}-{{ es_version }}` only when `es_version_locked.stdout|int == 0`, you run it only when there is existing elasticsearch package in versionlock (`yum versionlock list | grep -c {{es_package_name}}-{{es_version}}`), in this case, I don't understand why you need to run `yum versionlock delete 0:elasticsearch*`.

```
- name: RedHat - check elasticsearch version lock
  become: yes
  shell: yum versionlock list | grep -c {{es_package_name}}-{{es_version}}
  register: es_version_locked
  args:
    warn: false
  failed_when: False
  changed_when: False

- name: RedHat - lock elasticsearch version
  become: yes
  shell: yum versionlock delete 0:elasticsearch* ; yum versionlock add {{ es_package_name }}-{{ es_version }}
  args:
    warn: false
  when: es_version_lock and es_version_locked.stdout|int == 0
```

Reply:
I'm running version lock only when requested package is missing from `yum versionlock list`.

Maybe a little example will help:

ES package: `elasticsearch`
installed ES version: `7.0.1`
requested ES version: `7.1.1`
version lock enabled

`yum versionlock list` would return installed ES version:
```
Loaded plugins: fastestmirror, versionlock
0:elasticsearch-7.0.1-1.*
versionlock list done
```
so `yum versionlock list | grep -c elasticsearch-7.1.1`, because requested version is different from installed version, would return:
```
0
```
With this output we will set new version lock for requested version `7.1.1`, but running only `yum versionlock add elasticsearch-7.1.1` would fail with:
```
Excluding 1 update due to versionlock (use "yum versionlock status" to show it)
versionlock added: 0
```
It's necessary to remove existing version lock before creating new one.
I hope that's clear things up a little.

If that would help I can rearrange those tasks that `RedHat - unlock elasticsearch version` would run before `RedHat - lock elasticsearch version` (this time without `yum versionlock delete 0:elasticsearch*`) when upgrade is performed.